Allow acl in mqtt when client certificate is in use

GMagician opened the pull request:
Acl in mqtt needs user name if 'use_identity_as_username/use_subject_as_username' is not true. It also seems that login in mqtt doesn't accept an user with an empty password, then reenabled both

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [ ] The code change is tested and works with Tasmota core ESP8266 V.2.7.8
  - [x] The code change is tested and works with Tasmota core ESP32 V.3.1.1.250203
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

GMagician committed:
Allow acl in mqtt

Acl in mqtt needs user name if 'use_identity_as_username' is not true. It also seems that login in mqtt don't accept an user with an empty password, then reenabled both

s-hadinger commented:
The problem here is that Tasmota sets non-empty user/password for MQTT. This change would immediately break any existing configuration when using Client Certificate, and not setting user/password to empty strings.

I'm afraid there should be some option or additional setting to force user/password when using client cert

s-hadinger commented on tasmota/tasmota_xdrv_driver/xdrv_02_9_mqtt.ino:
```diff
@@ -1190,12 +1190,9 @@ void MqttReconnect(void) {
-  // re-assign private keys in case it was updated in between
+  // re-assign private key in case it was updated in between
   if (Mqtt.mqtt_tls) {
     if ((nullptr != AWS_IoT_Private_Key) && (nullptr != AWS_IoT_Client_Certificate)) {
```
As indicated in comment, this would break any existing client cert configuration where user/password have been left to default values. AWS IOT will refuse connection if you send non-empty user/password

GMagician replied:
ok then I think it will be useful to add a new define to let use custom certificate with mosquitto acl compatibility
What do you think about USE_MQTT_CUSTOM_CERT?

GMagician replied:
OT: I saw that web doesn't accept ';' char in password. Maybe some validation is required?

GMagician committed:
Add new define for mosquitto

Keep compatibility with AWS_IOT

GMagician commented:
> The problem here is that Tasmota sets non-empty user/password for MQTT. This change would immediately break any existing configuration when using Client Certificate, and not setting user/password to empty strings.
> 
> I'm afraid there should be some option or additional setting to force user/password when using client cert

I missed this....sorry. maybe may I recycle INCLUDE_LOCAL_CERT?

GMagician committed:
Better approach to custom client certificate

Don't duplicate defined check. Let USE_MQTT_AWS_IOT automatically set USE_MQTT_CLIENT_CERT and use latter in all code

GMagician commented:
Replaced by #22998
